File: solutions/python3/medium/queue-ticket-booking-system/solution-approach1.py
# ──────────────────────────────────────────────────
# Link        https://www.hackerrank.com/contests/ti27110-sri-sai-ram/challenges/queue-ticket-booking-system/problem?isFullScreen=true
# Problem     Queue - Ticket Booking System
# Difficulty  Medium
# Subdomain   N/A
# Platform    HackerRank
# Language    python3
# Status      Accepted
# Submitted   2026-07-29, 10:30 a.m.
# ──────────────────────────────────────────────────

class TicketQueue:
    def __init__(self, capacity):
        self.capacity = capacity
        self.queue = []

    def enqueue(self, name):
        if len(self.queue) >= self.capacity:
            print(f"Queue is full. Cannot add {name}")
        else:
            self.queue.append(name)

    def dequeue(self):
        if len(self.queue) == 0:
            print("Served Customer: Queue is empty. No customer to serve.")
        else:
            served = self.queue.pop(0)
            print(f"Served Customer: {served}")

    def display(self):
        if len(self.queue) == 0:
            print("Queue is empty.")
        else:
            print(f"Current Queue: {' '.join(self.queue)}")


if __name__ == "__main__":
    capacity = int(input().strip())
    ticket_system = TicketQueue(capacity)
    
    n_ops = int(input().strip())
    
    for _ in range(n_ops):
        line = input().strip().split()
        if not line:
            continue
            
        command = line[0]
        
        if command == "ENQUEUE":
            name = " ".join(line[1:])
            ticket_system.enqueue(name)
        elif command == "DEQUEUE":
            ticket_system.dequeue()
        elif command == "DISPLAY":
            ticket_system.display()

File: solutions/java15/medium/priority-queue-reordering-packages-in-a-delivery-center/solution-approach1.java
// ──────────────────────────────────────────────────
// Link        https://www.hackerrank.com/contests/ti27110-sri-sai-ram/challenges/priority-queue-reordering-packages-in-a-delivery-center/problem?isFullScreen=true
// Problem     Priority Queue - Reordering Packages in a Delivery Center
// Difficulty  Medium
// Subdomain   N/A
// Platform    HackerRank
// Language    java15
// Status      Accepted
// Submitted   2026-08-03, 10:44 p.m.
// ──────────────────────────────────────────────────

import java.util.PriorityQueue;
import java.util.Scanner;

public class Solution {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        if (!scanner.hasNextInt()) {
            return;
        }

        int n = scanner.nextInt();
        int[] arr = new int[n];

        for (int i = 0; i < n; i++) {
            arr[i] = scanner.nextInt();
        }

        if (!scanner.hasNextInt()) {
            return;
        }
        int k = scanner.nextInt();

        PriorityQueue<Integer> minHeap = new PriorityQueue<>();

        int index = 0;
        for (int i = 0; i <= k && i < n; i++) {
            minHeap.add(arr[i]);
        }

        for (int i = k + 1; i < n; i++) {
            arr[index++] = minHeap.poll();
            minHeap.add(arr[i]);
        }

        while (!minHeap.isEmpty()) {
            arr[index++] = minHeap.poll();
        }

        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + (i == n - 1 ? "" : " "));
        }
        System.out.println();

        scanner.close();
    }
}
